internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -667,19 +667,35 @@ func modifyMappedVirtualDisk(
 				}
 			}
 
+			if mvd.Encrypted {
+				if err := securityPolicy.EnforceEncryptedDeviceMountPolicy(mvd.MountPath); err != nil {
+					return errors.Wrapf(err,
+						"mounting encrypted scsi device controller %d lun %d onto %s denied by policy",
+						mvd.Controller,
+						mvd.Lun,
+						mvd.MountPath)
+				}
+			}
+
 			return scsi.Mount(mountCtx, mvd.Controller, mvd.Lun, mvd.MountPath,
 				mvd.ReadOnly, mvd.Encrypted, mvd.Options, mvd.VerityInfo)
 		}
 		return nil
 	case guestrequest.RequestTypeRemove:
 		if mvd.MountPath != "" {
-			err = securityPolicy.EnforceDeviceUnmountPolicy(mvd.MountPath)
-			if err != nil {
-				return errors.Wrapf(err, "unmounting scsi device at %s denied by policy", mvd.MountPath)
-			}
+			if mvd.ReadOnly {
```

Comment:
Doesn't this logic change behavior? If ReadOnly and Encrypted is false, there is no policy enforcement.

Reply:
I think we just missed this during review, but currently we don't have any enforcement for non-readonly devices: https://github.com/microsoft/hcsshim/blob/main/internal/guest/runtime/hcsv2/uvm.go#L657-L668. I think I missed the unmount path during review.